### PR TITLE
[SYCL][DOC] Clarify device global error conditions

### PR DESCRIPTION
**Pull request by gmlueck**
Clarify the behavior of `handler::copy()` and `handler::memcpy()` in
two error cases w.r.t `device_global` variables.

### CODE REVIEW

**Review by mkinsner — APPROVED** [review]

**Comment by gmlueck** [comment]
@bader: Can you merge this now that it is approved?

**Comment by bader** [comment]
> @bader: Can you merge this now that it is approved?

Yes, after CI checks unblock merge button.

**Pull request merged**
